src/main/java/sigma/Storage.java

```diff
@@ -0,0 +1,153 @@
+package sigma;
+
+import sigma.exception.SigmaException;
+import sigma.task.DeadlineTask;
+import sigma.task.EventTask;
+import sigma.task.Task;
+import sigma.task.ToDoTask;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the storage of tasks.
+ * Handles the reading and writing of tasks to a file.
+ * The file is stored in the file path specified by the user.
+ * The file is created if it does not exist.
+ * The file is read and written to in the format:
+ * [task type] | [status] | [description] | [date]
+ * where task type is T for ToDoTask, D for DeadlineTask, and E for EventTask.
+ * where status is X if the task is done, and a space if the task is not done.
+ * where date is the date of the task in the format "MMM d yyyy, HH:mm".
+ */
+public class Storage {
+
+    private String filePath;
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        load(filePath);
+    }
+
+    /**
+     * Writes the tasks to the file.
+     * @param data File to write to.
+     * @param items List of tasks to write to the file.
+     */
+    public void write(File data, ArrayList<Task> items) {
+        try {
+            FileWriter writer = new FileWriter(data);
+            for (Task item : items) {
+                writer.write(String.format("%s | %s | %s | %s \n",
+                        item.getTaskType(), item.getStatusString(), item.getDesc(),
+                        item instanceof DeadlineTask
+                                ? ((DeadlineTask) item).getDate()
+                                : item instanceof EventTask
+                                ? ((EventTask) item).getFrom() + " - " + ((EventTask) item).getTo()
+                                : ""));
```

Comment:
I found this nested ternary operator very hard to read, it might be more easy to understand if you refactor this logic to earlier lines using `if-else` statements

Reply:
You're right, I'll change this!

Reply:
Changed to:
``` java
for (Task item : items) {
                String date;
                if (item instanceof DeadlineTask) {
                    date = ((DeadlineTask) item).getDate();
                } else if (item instanceof EventTask) {
                    date = ((EventTask) item).getFrom() + " - " + ((EventTask) item).getTo();
                } else {
                    date = "";
                }
                writer.write(String.format("%s | %s | %s | %s \n",
                        item.getTaskType(), item.getStatusString(), item.getDesc(), date));
            }
            writer.close();
```